server/data-service/src/main/java/edu/illinois/ncsa/incore/service/data/utils/GeoserverRestApi.java

```diff
@@ -131,6 +134,192 @@ public boolean postFileToGeoserver(String apiUrl, File inData, String contType)
         }
     }
 
+    /**
+     * upload file to geoserver
+     *
+     * @param store
+     * @param inFile
+     * @param inExt
+     * @return
+     */
+    public boolean uploadToGeoserver(String store, File inFile, String inExt) {
+        String fileName = FilenameUtils.getBaseName(inFile.getName());
+
+        // check if workspace exists
+        boolean created = createWorkspace(this.geoserverUrl, GEOSERVER_WORKSPACE);
+        boolean published = false;
+
+        if (inExt.equalsIgnoreCase("shp")) {
+            String restUrl = this.geoserverUrl + "/rest/workspaces/" + GEOSERVER_WORKSPACE + "/datastores/" + store + "/file.shp";
+            published = this.postFileToGeoserver(restUrl, inFile, "zip");
+        } else if (inExt.equalsIgnoreCase("gpkg")) {
+            // geopackage should have different steps due to its renaming convention in IN-CORE
+            // since IN-CORE uses the dataset id as a file name in geoserver, the geopackage name needs to be renamed to dataset id
+            // to rename, the POST action should create datastore first with the geopackage data,
+            // then create layer with xml to control the name to be changed to dataset id.
+            // if you don't need to rename, set renameData boolean in following line to false
+            Boolean renameData = true;
```

Comment:
then it will just do the regular upload and it will upload the geopackage as the geopackage file / data table name instead of dataset id, and it will  lose the uniqueness and the connection between the dataset entry

Reply:
Resovling this conversation since we talked about refactoring that part in another task.